src/crucible/aws/ecs/task_definition.clj

```diff
@@ -0,0 +1,40 @@
+(ns crucible.aws.ecs.task-definition
+  "Resources in AWS::ECS::TaskDefinition"
+  (:require [clojure.spec.alpha :as s]
+            [crucible.resources :refer [spec-or-ref defresource]]
+            [crucible.aws.ecs.container-definition :as cd]))
+
+(s/def ::container-definitions (s/coll-of ::cd/container-definition :kind vector?))
+
+(s/def ::cpu (spec-or-ref string?))
+
+(s/def ::execution-role-arn (spec-or-ref string?))
+
+(s/def ::family (spec-or-ref string?))
+
+(s/def ::memory (spec-or-ref string?))
+
+(s/def ::network-mode (spec-or-ref string?))
```

Comment:
Could be more specific:

(s/def ::network-mode #{"bridge" "host" "awsvpc" "none"})